### PR TITLE
Version 2: Updates teams to use dedicated spaces

### PR DESCRIPTION
**Pull request by ahamilton55**
- Updates team secrets to now use dedicates spaces instead of providing secrets to each team member individually. This allows secrets to persist if all members of a team rotate out.
- Updates Vault policy generation to allow for generating access to team spaces.
- Allows for case-insensitive lookups without breaking backwards compatibility for previously created secrets.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -27,7 +27,7 @@
```
Comment (hankjacobs) [inline Gopkg.toml]: Maybe convert to Go 1.11 and modules?

File: cmd/get.go [hunk 2]
```diff
@@ -6,8 +6,14 @@ import (
 	"github.com/spf13/cobra"
 )
 
+var (
+	team string
```
Comment (hankjacobs) [inline cmd/get.go:10]: I'm not a big fan of global state (even for flags). I know cobra encourages this pattern, though.

File: cmd/root.go [hunk 3]
```diff
@@ -22,12 +22,17 @@ var (
 
 	dirState      directory.Backend
 	storageClient storage.Backend
+
+	updateCache bool
```
Comment (hankjacobs) [inline cmd/root.go:26]: Ditto my above comment

File: pkg/directory/gh.go [hunk 4]
```diff
@@ -79,6 +79,12 @@ func (g *GH) getMembersAndTeams() error {
 		update = true
 	}
 
+	activeMembershipsFile := path.Join(cacheDir, "active-memberships")
```
Comment (hankjacobs) [inline pkg/directory/gh.go]: `filepath.Join`

**Review by hankjacobs — CHANGES_REQUESTED** [review]
One change and some nitpicks for consideration

File: cmd/get.go (hunk 2 above)
Comment (ahamilton55) [reply cmd/get.go:10]: Not a fan either but these are limited to the cmd package and aren't exported. There is the possibility of adding them into the commands but that requires calling flag.Parse() in ever command function .

File: Gopkg.toml (hunk 1 above)
Comment (ahamilton55) [reply Gopkg.toml]: Done and updated CircleCI

**Review by hankjacobs — APPROVED** [review]

**Pull request merged**